src/Core/Vipr.Core/CodeModel/OdcmCapabilities.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vipr.Core.CodeModel
+{
+    public class OdcmCapabilities
+    {
+        public bool Insertable { get; set; }
```

Comment:
@MrTomWhite I feel the original model adds an extra layer in converting the raw annotations into OdcmCapability. Also in this case the raw annotation (like insertrestriction) properties do not map directly to OdcmCapbility class. 

Also OdataCapabiltiesReader right now dispatches the parsing logic to a list of parsers which can parse the annotations any way they want.